www/main/navigator/misc/components/favoritePane.jsx

```diff
@@ -0,0 +1,180 @@
+  const [errorMessage, setError] = useState('');
+
+  const initialFetch = async () => {
+    const onlineValue = await isOnline();
+    if (onlineValue) {
+      if (userToken) {
+        setError('');
+        setFetching(true);
+        const data = await fetchFavShabad(userToken);
+        setFetching(false);
+        setFavShabad([...data]);
+      } else {
+        setError(i18n.t('FAV_SHABAD.LOGGED_OUT'));
+        setFavShabad([]);
+      }
+    } else {
+      setError(i18n.t('FAV_SHABAD.INTERNET_ERR'));
+    }
+  };
+
+  useEffect(() => {
+    initialFetch();
+  }, [userToken]);
+
+  const deleteFromFav = (inputElement) => {
+    const favShabadIndex = favShabad.findIndex(
+      (element) => element.shabad_id === inputElement.shabadId,
+    );
+    removeFromFav(inputElement.shabadId, userToken);
+
+    if (favShabadIndex >= 0) {
+      favShabad.splice(favShabadIndex, 1);
+      setFavShabad([...favShabad]);
+    }
+  };
+
+  const openShabadFromFav = (shabadId, verseId) => {
+    if (singleDisplayActiveTab !== 'shabad') {
+      setSingleDisplayActiveTab('shabad');
+    }
+    if (shabadId !== activeShabadId) {
+      setActiveShabadId(shabadId);
+
+      if (verseId !== activeVerseId) {
+        setActiveVerseId(verseId);
+      }
+
+      if (verseId !== initialVerseId) {
+        setInitialVerseId(verseId);
+      }
+
+      if (verseId !== homeVerse) {
+        setHomeVerse(verseId);
+      }
+
+      if (versesRead !== []) {
+        setVersesRead([]);
+      }
+
+      if (isSundarGutkaBani) {
+        setIsSundarGutkaBani(false);
+      }
+
+      if (isCeremonyBani) {
+        setIsCeremonyBani(false);
+      }
+    }
+  };
+
+  useEffect(() => {
+    const fetchData = async () => {
+      const promises = favShabad.map(async (element) => {
+        const elementDate = new Date(element.created_at);
+        const dateString = elementDate.toLocaleDateString('en-us', {
+          day: 'numeric',
+          year: 'numeric',
+          month: 'short',
+        });
+        const timeString = elementDate.toLocaleTimeString('en-us', {
+          hour: 'numeric',
+          minute: 'numeric',
+        });
+        const row = await banidb.getVerse(element.shabad_id, element.verse_id);
+        return {
+          date: dateString,
+          time: timeString,
+          shabadId: element.shabad_id,
+          verseId: element.verse_id,
+          verse: row,
+          id: element.id,
+        };
+      });
+      Promise.all(promises).then((data) => {
+        setParsedFav(data);
+      });
+    };
+    fetchData();
+  }, [favShabad]);
+
+  return (
+    <div className={`fav-results ${className}`}>
+      <p>{errorMessage}</p>
+      {isFetching && <div className="sttm-loader" />}
+      {parsedFav.map((element, index) => {
+        const { shabadId, verseId, date, time, verse, id } = element;
+        return (
+          <div className="fav-shabad-container" key={`fav-shabad-${index}`}>
```

Comment:
This piece of code should be refactored. One way with more readability could be the below one.

```jsx
parsedFav.map(({ shabadId, verseId, date, time, verse, id }, index) => (
          <div className="fav-shabad-container" key={`fav-shabad-${index}`}>
```

Reply:
We use the entire object `element` at line 165 of this file, so need that as parameter here.